### PR TITLE
Update containers: Scafacos, 4.0.2

### PR DESCRIPTION
**Pull request by jngrad**
* Transfer Scafacos from Ubuntu 18 Python 2 to Ubuntu 18 Python 3 (espressomd/espresso#2694)
* Test 4.0.x-specific containers against the 4.0 branch, which was recently fast-forwarded to 4.0.2

### CODE REVIEW

**Comment by jngrad** [comment]
Notes:
Ubuntu warning `Download is performed unsandboxed as root as file '/var/cache/apt/archives/partial/liblocale-gettext-perl_1.07-3build2_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)` is a [known bug](https://bugs.launchpad.net/ubuntu/+source/synaptic/+bug/1522675) unrelated to Docker.
Scafacos warning `make[1]: [make-python-files] Error 1 (ignored)` was already present in the Python 2 container ([ubuntu:18.04 logfile](https://gitlab.icp.uni-stuttgart.de/espressomd/docker/-/jobs/92460)).

**Comment by mkuron** [comment]
Something is wrong with the exotic architecture containers' build process:
```
$ sh maintainer/docker_build.sh
Command failed
>cd docker/ubuntu
>test -f Dockerfile-s390x
Job succeeded
```
Not sure if this was caused by your changes or is an older issue. By the way, these containers should also use Python 3 now.

File: docker/ubuntu-python3/build-and-install-scafacos.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+set -e
+
+cd /tmp
+git clone --recursive git://github.com/scafacos/scafacos --branch dipoles
+cd scafacos
+./bootstrap
+./configure --enable-shared \
+	--with-internal-pfft --with-internal-pnfft \
+	--enable-fcs-solvers=direct,pnfft,p2nfft,p3m \
+	--disable-fcs-fortran --enable-fcs-dipoles
```
Comment (fweik) [inline docker/ubuntu-python3/build-and-install-scafacos.sh]: Do you still need to install `gfortran` if fortran is  disabled?

Comment (jngrad) [reply docker/ubuntu-python3/build-and-install-scafacos.sh]: I've tried just now and apparently we need an MPI Fortran compiler nonetheless:
```
checking for mpic++... mpic++
checking whether we are using the GNU C++ compiler... yes
checking whether mpic++ accepts -g... yes
checking dependency style of mpic++... gcc3
checking for function MPI::Init... yes
checking for mpi.h... yes
checking for mpif95... no
checking for mpxlf95_r... no
checking for mpxlf95... no
checking for ftn... no
checking for mpif90... mpif90
checking whether we are using the GNU Fortran compiler... no
checking whether mpif90 accepts -g... no
checking for function MPI_INIT... no
checking for function MPI_INIT in -lmpichf90... no
checking for function MPI_INIT in -lfmpi... no
checking for function MPI_INIT in -lfmpich... no
configure: error: in `/tmp/scafacos':
configure: error: ScaFaCoS requires an MPI Fortran compiler.
See `config.log' for more details
The command '/bin/sh -c bash /tmp/build-and-install-scafacos.sh' returned a non-zero code: 1
```

Comment (jngrad) [reply docker/ubuntu-python3/build-and-install-scafacos.sh]: `--disable-fcs-fortran` means `Disable Fortran interface and tests`, but scafacos includes a couple Fortran libraries (`scafacos/lib/pepc`, `scafacos/lib/fmm`), which would explain the error message when `gfortran` is not installed.

Comment (fweik) [reply docker/ubuntu-python3/build-and-install-scafacos.sh]: The wonders of autotools...

**Comment by RudolfWeeber** [comment]
Let's live with installing gfortran. The Scafacos build system is rather complex.

**Pull request merged**